### PR TITLE
Headings Changes and Heading Component

### PR DESCRIPTION
**Pull request by ElectronicBlueberry**
This PR implements the suggested changes from #14585

Following changes have been implemented:

## Introduce heading size classes

Following global utility classes have been implemented to size headings:

 - `h-xl` - Extra large headings
 - `h-lg` - Large headings (most main headings should have this size)
 - `h-md` - Medium headings
 - `h-sm` - Small headings (many sub-headings have this size)
 - `h-text` - Text sized headings

## Add heading component

A new versatile heading component has been added, which can be used for normal, or decorated headings.

It has several properties to easily configure the heading:

* `h1 ... h6` defines the headings level
* `size=["large", "medium", "small", "text"]` sets the headings size class
* `bold`
* `inline`
* `separator` adds a separating line decoration
* `icon="..."` adds a font-awesome icon decoration to the heading

## Restructuring existing Headings

Most headings in vue files have been restructured to follow best practices. I tried to be thorough, but it is very likely I missed a few.

## Various changes

Some headings have been resized for more consistency.
The new `<Heading>` component has been introduced in some places, mostly as an usage example.

## Heading component example

The following code:
```vue
<Heading>Default Heading</Heading>
<Heading h2 size="large">Large H2 Heading</Heading>
<Heading h3 separator size="medium">Medium H3 Heading with Separator</Heading>
<Heading h3 size="medium" icon="fa-angle-right">Heading with Icon</Heading>
<Heading h4 size="small" icon="fa-angle-double-right">Small Heading with Icon</Heading>
<Heading h2 bold size="medium">Medium Bold H2</Heading>
<span> <Heading h3 bold inline size="text">Inline Text Sized Bold H3</Heading> next to some text.</span>
<Heading h3 separator bold size="medium">Bold H3 with Separator</Heading>
<Heading h4 bold size="medium" icon="fa-question">Bold Medium H4 with Icon</Heading>
<Heading h4 size="small" icon="far fa-question-circle">Small H4 with Icon</Heading>
<Heading h3 separator size="small">Small H3 with Separator</Heading>
```

results in:

image

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by guerler** [comment]
Thank you for working on this, I noticed a minor issue, title panel font sizes differ: 
image

**Comment by ElectronicBlueberry** [comment]
I'm still in the process of editing the heading styles. I'll make sure to keep the history heading size the same as the Toolbox, once I adjust the histories headings

File: client/src/style/scss/ui.scss [hunk 1]
```diff
@@ -473,6 +473,39 @@ $ui-margin-horizontal-large: $margin-v * 2;
...
+}
+
+.h-sm {
+    font-size: $h4-font-size;
+}
+
+.h-text {
+    font-size: $font-size-base;
+}
+
```
Comment (guerler) [inline client/src/style/scss/ui.scss:496]: This decoupling of header font-sizes from header-tags which in are then used semantically is awesome. Thank you also for the documentation.

**Comment by guerler** [comment]
I tested this and it worked well. This decoupling of header font-sizes from header-tags which are then used semantically is awesome. Thank you also for the documentation.

File: lib/galaxy/webapps/galaxy/buildapp.py [hunk 2]
```diff
@@ -271,6 +271,7 @@ def app_pair(global_conf, load_app_kwds=None, wsgi_preflight=True, **kwargs):
     webapp.add_client_route("/interactivetool_entry_points/list")
     webapp.add_client_route("/libraries{path:.*?}")
     webapp.add_client_route("/storage{path:.*?}")
+    webapp.add_client_route("/headings_demo")
```
Comment (guerler) [inline lib/galaxy/webapps/galaxy/buildapp.py]: Where is this route handled?

Comment (dannon) [reply lib/galaxy/webapps/galaxy/buildapp.py]: It was temporary, for a test / demo that was removed -- just needs to be deleted. (https://github.com/galaxyproject/galaxy/pull/14622/commits/cfdf67056046fe679aa6cfa41e90c0f7da8d370f)

```suggestion
```

**Review by guerler — APPROVED** [review]

**Review by dannon — COMMENTED** [review]
Very nice work here, thanks @ElectronicBlueberry!

**Pull request merged**